proposals/22-04-localize-command.md

```diff
@@ -61,48 +65,49 @@ where the arguments are:
 
 * `target`: a directory with a top-level kustomization file that kustomize will localize; can be a path to a local
   directory or a url to a remote directory
-* `scope`: `target` or a directory that contains `target`
-* `newDir`: destination of the localized copy of `target`
-    * if `target` is local, `newDir` must be a directory name, as it will be located in the same directory as `target`
-      to preserve relative path references
-    * if `target` is remote, `newDir` must be a directory path
+* `scope`: optional root directory, files outside which kustomize is not allowed to copy and localize; if not specified,
+  takes on value of `target`
+* `newDir`: destination directory of the localized copy of `target`
+
+The command creates a copy of the `target` kustomization and the local files that `target` references at `newDir`. We
+define the "files that `target` references" as:
 
-The command creates a copy of `target` at `newDir` in which each kustomization file, from the top-level to any
-recursively referenced, has local paths to downloaded files instead of remote references for the following kustomization
-fields:
+* kustomization files that `target` directly or transitively references
+* configuration files that referenced kustomization files reference
+* exec binaries of referenced KRM functions
 
-* `resources`
-* `components`
-* `bases`
-* `openapi:`  
-  &nbsp; &nbsp; `paths`
+Here, configuration file means a non-kustomization yaml file. The command only copies referenced files that reside
+inside `scope`.
 
-A new `localized-files` directory holds the downloaded files (or directory)
-at:
+The command localizes the copy of `target` at `newDir` by downloading all remote files that `target` references. The
+downloaded files placed in a new `localized-files` directory next to the file that referenced the downloaded files.
+Inside `localized-files`, the downloads are located on path:
 
 <pre>
 <ins>remote-host</ins> / <ins>organization</ins> / <ins>repo</ins> / <ins>version</ins> / <ins>path-to-file-in-repo</ins>
 </pre>
 
-Each `localized-files` directory is located in the same directory as the kustomization file that referenced the
-downloaded files.
-
-To help ensure that `newDir` is a clean copy, the command overwrites every absolute path into `target` to point
-to `newDir` before processing the path.
+The command rewrites remote references in `newDir` to the local paths of the downloaded files. To help ensure
+that `newDir` is a clean copy, the command additionally overwrites absolute path references into `target` to point
```

Comment:
I suggest adding that we won't handle absolute paths in alpha. I'm not even sure where we allow absolute paths (if there are any).

Reply:
Will do. I added this because I saw [this comment for `Resources`](https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/kustomization.go#L106) and `Components`.